org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandlerTest.java

```diff
@@ -145,6 +148,18 @@ public void testJavaImportExclusions() throws Exception {
 		assertEquals(TEST_EXCLUSIONS, prefs.getJavaImportExclusions().get(0));
 	}
 
+	@Test
+	public void testWatchers() throws Exception {
+		ClientPreferences mockCapabilies = mock(ClientPreferences.class);
+		when(mockCapabilies.isWorkspaceChangeWatchedFilesDynamicRegistered()).thenReturn(Boolean.TRUE);
+		when(preferenceManager.getClientPreferences()).thenReturn(mockCapabilies);
+		newEmptyProject();
+		List<FileSystemWatcher> watchers = projectsManager.registerWatchers();
+		assertNotNull(watchers);
+		assertTrue(watchers.size() == 1);
+		assertEquals(watchers.get(0).getGlobPattern(), ResourcesPlugin.getWorkspace().getRoot().getLocation().toString() + "/TestProject/src/**");
+	}
+
```

Comment:
can you check a Maven or Gradle project with src/main/resources gets the expected watchers?